docs/concepts.rst

```diff
@@ -309,87 +309,85 @@ The :class:`~pystac.validation.JsonSchemaSTACValidator` takes a :class:`~pystac.
 
-Notice the ``eo`` property on ``ext`` - this utilizes the `__getattr__ <https://docs.python.org/3/reference/datamodel.html#object.__getattr__>`_ method to delegate the property name to the ``__getitem__`` method, so we can access any registered extension as if it were a property on ``ext``.
+Accessing Extension Functionality
+---------------------------------
 
-Extensions wrap the objects they extend. Extensions hold
-no values of their own, but instead use Python `properties <https://docs.python.org/3/library/functions.html#property>`_
-to directly modify the values of the objects they wrap.
+Extension functionality is encapsulated in classes that are specific to the STAC
+Extension (e.g. Electro-Optical, Projection, etc.) and STAC Object
+(:class:`~pystac.Collection`, :class:`pystac.Item`, or :class:`pystac.Asset`). All
+classes that extend these objects inherit from
+:class:`pystac.extensions.base.PropertiesExtension`, and you can use the
+``ext`` method on these classes to extend an object.
 
-Any object that is returned by extension methods therefore also wrap components of the STAC objects.
-For instance, the ``LabelClasses`` holds a reference to the original ``Item``'s ``label:classes`` property, so that
-modifying the ``LabelClasses``
-properties through the setters will modify the item properties directly. For example:
+For instance, to extend an item with the :stac-ext:`Electro-Optical Extension <eo>`
+you would use :meth:`EOExtension.ext <pystac.extensions.eo.EOExtension.ext>` as
+follows:
 
 .. code-block:: python
 
-    from pystac.extensions import label
+   import pystac
+   from pystac.extensions.eo import EOExtension
 
-    label_classes = item.ext.label.label_classes
-    label_classes[0].classes.append("other_class")
-    assert "other_class" in item.properties['label:classes'][0]['classes']
+   item = Item(...)  # See docs for creating an Item
+   eo_ext = EOExtension.ext(item)
 
-Because these objects wrap the object's dictionary, the __init__ methods need to take the
-``dict`` they wrap. Therefore to create a new object, use the class's `.create` method, for example:
+This extended instance now gives you access to the properties defined in that extension:
 
 .. code-block:: python
 
-   item.ext.label.label_classes = [label.LabelClasses.create(['class1', 'class2'], name='label')]
-
-An `apply` method is available in extension wrappers and any objects that they return. This allows
-you to pass in property values pertaining to the extension. These will require arguments for properties
-required as part of the extension specification and have `None` default values for optional parameters:
-
-.. code-block:: python
+   eo_ext.bands
+   eo_ext.cloud_cover
 
-   eo_ext = item.ext.eo
-   eo_ext.apply(0.5, bands, cloud_cover=None) # Do not have to specify cloud_cover
+See the documentation for each extension implementation for details on the supported
+properties and other functionality.
```

Comment:
Another topic worth discussing in this section or elsewhere is the mutability of the underlying STACObject when using extensions. E.g. in the above example, if I set `eo_ext.cloud_cover = 0.1`, then the properties of the wrapped Item is modified. This includes extensions objects that are generated from the extensions; if I were to say do

```
eo_ext.bands[0].common_name = "purple"
```

then the underlying Item or Collection properties would be modified to change the common name of the first band. Users should be aware and careful about the highly mutable nature of PySTAC in general and extensions in particular

Reply:
Good call. I added a note to this section as well as an example of that mutability.